src/HealthKit/HKCategoryValueSleepAnalysisAsleep.cs

```diff
@@ -0,0 +1,56 @@
+// HKCategoryValueSleepAnalysisAsleep
+
+//
+// HKCategoryValueSleepAnalysisAsleep.cs:
+//
+// Authors:
+//  TJ Lambert (TJ.Lambert@microsoft.com
+//
+// Copyright 2022 Microsoft Corp.
+//
+
+#nullable enable
+
+#if !TVOS
+
+using System;
+using Foundation;
+using ObjCRuntime;
+using System.Runtime.InteropServices;
+using System.Collections.Generic;
+
+#if !NET
+using NativeHandle = System.IntPtr;
+#endif
+
+namespace HealthKit
+{
+	public partial class HKCategoryValueSleepAnalysisAsleep {
+
+#if NET
+		[SupportedOSPlatform ("ios16.0")]
+		[SupportedOSPlatform ("macos13.0")]
+		[SupportedOSPlatform ("maccatalyst16.0")]
+		[UnsupportedOSPlatform ("tvos")]
+#else
+		[iOS (16,0), Mac (13,0), Watch (9,0), NoTV, MacCatalyst (16,0)]
+#endif // NET
+		[DllImport (Constants.HealthKitLibrary)]
+		static extern /* HKCategoryValueSleepAnalysis */ NativeHandle HKCategoryValueSleepAnalysisAsleepValues ();
+
+#if NET
+		[SupportedOSPlatform ("ios16.0")]
+		[SupportedOSPlatform ("macos13.0")]
+		[SupportedOSPlatform ("maccatalyst16.0")]
+		[UnsupportedOSPlatform ("tvos")]
+#else
+		[iOS (16,0), Mac (13,0), Watch (9,0), NoTV, MacCatalyst (16,0)]
+#endif // NET
+		public static NSSet<NSNumber> GetAsleepValues ()
+		{
+			return Runtime.GetNSObject<NSSet<NSNumber>> (HKCategoryValueSleepAnalysisAsleepValues ())!;
+		}
```

Comment:
This method should not be returning a NSNumber, but a HKCategoryValueSleepAnalysis, if you look at the headers we have:
```
+/// Returns the set of `HKCategoryValueSleepAnalysis` values that are considered to be asleep.
```
Where HKCategoryValueSleepAnalysis is 
```
typedef NS_ENUM(NSInteger, HKCategoryValueSleepAnalysis) {
```
You will need to update the manual tests.

Reply:
Alex and I talked about this one and kept it as is, but I will give it a shot returning the strong types